scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -1326,3 +1387,71 @@ def f(x):
         with pytest.raises(ValueError, match='One of the integrality'):
             DifferentialEvolutionSolver(f, bounds=bounds, polish=False,
                                         integrality=integrality)
+
+    def test_vectorized(self):
+        def quadratic(x):
+            return np.sum(x**2)
+
+        def quadratic_vec(x):
+            return np.sum(x**2, axis=0)
+
+        # A vectorized function needs to accept (len(x), S) and return (S,)
+        with pytest.raises(RuntimeError, match='The vectorized function'):
+            differential_evolution(quadratic, self.bounds,
+                                   vectorized=True, updating='deferred')
+
+        # vectorized overrides the updating keyword, check for warning
+        with warns(UserWarning, match="differential_evolution: the 'vector"):
+            differential_evolution(quadratic_vec, self.bounds,
+                                   vectorized=True)
+
+        # vectorized defers to the workers keyword, check for warning
+        with warns(UserWarning, match="differential_evolution: the 'workers"):
+            differential_evolution(quadratic_vec, self.bounds,
+                                   vectorized=True, workers=map,
+                                   updating='deferred')
+
+        def rosen_vec(x):
```

Comment:
For now it's fine, but we could as well update `rosen`. I can see user trying to understand the new option with `rosen`

Reply:
This was a remnant of a previous approach, `rosen` is already vectorised for what we need. I've made a change, keeping this function as a wrapper to remember the number of function calls.